core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -2209,6 +2284,15 @@ object Parser {
       override def parseMut(state: State): Int = state.offset
     }
 
+    case object WithCaret extends Parser0[Caret] {
+      override def parseMut(state: State): Caret = {
+        val (row, col) = state.locmap
+          .toLineCol(state.offset)
+          .getOrElse(throw new IllegalStateException("the offset is larger than the input size"))
```

Comment:
this will throw when you are at the EOF right?

I think we need to fix that somehow.

I guess we can say that EOF should return `toLineCol(eof - 1).map { case (row, col) => (row, col + 1) }`

What do you think?

Reply:
Yes. I'll take a look at how I can solve that first.

Reply:
Added `GenT` for spanning combinators but not yet added to the list in `gen(0)`. Will add after this is solved.

Reply:
#286 fixes this issue.